plugins/sinks/gcs/gcs.go

```diff
@@ -0,0 +1,161 @@
+package gcs
+
+import (
+	"context"
+	_ "embed"
+	"encoding/base64"
+	"strings"
+	"time"
+	"fmt"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/odpf/meteor/models"
+	assetsv1beta2 "github.com/odpf/meteor/models/odpf/assets/v1beta2"
+	"github.com/odpf/meteor/plugins"
+	"github.com/odpf/meteor/registry"
+	"github.com/odpf/salt/log"
+	"github.com/pkg/errors"
+)
+
+//go:embed README.md
+var summary string
+
+type Config struct {
+	ProjectID            string `mapstructure:"project_id" validate:"required"`
+	Path                 string `mapstructure:"path" validate:"required"`
+	ObjectPrefix         string `mapstructure:"object_prefix"`
+	ServiceAccountJSON   string `mapstructure:"service_account_json"`
+	ServiceAccountBase64 string `mapstructure:"service_account_base64"`
+}
+
+var info = plugins.Info{
+	Description: "saves data in google cloud storage bucket",
+	Summary:     summary,
+	SampleConfig: heredoc.Doc(`
+	project_id: google-project-id
+	path: bucket_name/target_folder
+	object_prefix : github-users
+	service_account_base64: ____base64_encoded_service_account____
+    service_account_json: |-
+      {
+       "type": "service_account",
+       "private_key_id": "xxxxxxx",
+       "private_key": "xxxxxxx",
+       "client_email": "xxxxxxx",
+       "client_id": "xxxxxxx",
+       "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+       "token_uri": "https://oauth2.googleapis.com/token",
+       "auth_provider_x509_cert_url": "xxxxxxx",
+       "client_x509_cert_url": "xxxxxxx"
+     }
+	`),
+	Tags: []string{"gcs", "sink"},
+}
+
+type Sink struct {
+	plugins.BasePlugin
+	logger log.Logger
+	writer Writer
+	config Config
+}
+
+func New(logger log.Logger) plugins.Syncer {
+	s := &Sink{
+		logger: logger,
+	}
+	s.BasePlugin = plugins.NewBasePlugin(info, &s.config)
+
+	return s
+}
+
+func (s *Sink) Init(ctx context.Context, config plugins.Config) (err error) {
+	if err = s.BasePlugin.Init(ctx, config); err != nil {
+		return err
+	}
+
+	if err := s.validateServiceAccountKey(); err != nil {
+		return err
+	}
+
+	bucketname, objectname := s.resolveBucketandObjectNames()
+
+	if s.writer, err = newWriter(ctx, []byte(s.config.ServiceAccountJSON), bucketname, objectname); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *Sink) validateServiceAccountKey() error {
+	if s.config.ServiceAccountBase64 == "" && s.config.ServiceAccountJSON == "" {
+		return errors.New("credentials are not specified, failed to create client")
+	}
+
+	if s.config.ServiceAccountBase64 != "" {
+		serviceAccountJSON, err := base64.StdEncoding.DecodeString(s.config.ServiceAccountBase64)
+		if err != nil || len(serviceAccountJSON) == 0 {
+			return errors.Wrap(err, "failed to decode base64 service account")
+		}
+		s.config.ServiceAccountJSON = string(serviceAccountJSON)
+	}
+	return nil
+}
+
+func (s *Sink) resolveBucketandObjectNames() (string, string) {
+	dirs := strings.Split(s.config.Path, "/")
+	bucketname := dirs[0]
```

Comment:
A notation pretty common in object storage is `gcs://bucketname` or `s3://bucketname` I guess this line would fail here. Can we use `url.Parse(...)`?